app/src/routes/api/v1/worldbank.router.js

```diff
@@ -30,6 +30,11 @@ class WbIndexRouter {
         logger.info('Registering dataset with data', ctx.request.body);
         try {
             await WBIndexService.register(ctx.request.body.connector, ctx.request.body.userId);
+        } catch (e) {
```

Comment:
So this is the only change: if we fail to create the metadata, we should bail with error, and not try to patch the metadata.

Reply:
reverted